caesars-palace/blocks/slider-card/slider-card.js

```diff
@@ -1,42 +1,75 @@
 export default function decorate(block) {
   const cardWrapper = document.createElement('div');
   cardWrapper.classList.add('card-wrapper');
-  block.querySelectorAll('div.slider-card > div').forEach((div) => {
+
+  function isADesktop() {
+    const mediaDesktop = window.matchMedia('only screen and (min-width: 769px)');
+    return mediaDesktop.matches;
+  }
+
+  // default value is 3
+  let numberOfCardsDisplayed = 3;
+  block.querySelectorAll('div.slider-card > div').forEach((div, idx) => {
+    // We do not need to add a class to the first div as it includes the number of
+    // cards we wish to display
+    if (idx === 0) {
+      numberOfCardsDisplayed = div.children[0].innerHTML;
+      div.style.display = 'none';
+      if (isADesktop()) {
+        const sliderCard = block;
+        sliderCard.style = `width: calc((100%/${numberOfCardsDisplayed} - 10px)); gap: 10px;`;
+      }
+      return;
+    }
     div.classList.add('card');
-    const imageDiv = div.children[0];
-    imageDiv.children[0].children[3].classList.add('card-image');
-    div.children[1].classList.add('short-description');
-    div.children[2].classList.add('long-description');
+    let index = 0;
+    if (div.getElementsByTagName('picture').length > 0) {
+      const imageDiv = div.children[index];
+      imageDiv.children[index].children[3].classList.add('card-image');
+      index += 1;
+    }
+    div.children[index].classList.add('short-description');
+    div.children[index + 1].classList.add('long-description');
+
     const closeButton = document.createElement('div');
     closeButton.classList.add('close-button');
     closeButton.classList.add('hide');
-    div.insertBefore(closeButton, div.children[2]);
+    div.insertBefore(closeButton, div.children[index + 1]);
   });
 
-  const shortDescriptionDivs = document.querySelectorAll('.short-description');
+  const shortDescriptionDivs = block.querySelectorAll('.short-description');
   shortDescriptionDivs.forEach((div) => {
+    let showMore;
     const title = div.children[0].children[0];
-    title.classList.add('title');
-    const showMore = div.children[1].children[0];
-    if (showMore) {
-      showMore.classList.add('show-more');
+    title?.classList.add('title');
+    if (div.children.length >= 3) {
+      [, , showMore] = div.children;
+      const discount = div.children[1];
+      discount.classList.add('discount');
+    } else {
+      [, showMore] = div.children;
+    }
+    if (showMore && showMore.children.length > 0) {
+      showMore.children[0].classList.add('show-more');
     }
```

Comment:
I think this could be simplifed too. **Discount** is too specific to one use case. Instead the author could just use an `h5` or `h6` if they want a sub title.

Also, could it be posible for the `...more` text to be automatically added to the short description instead of forcing the author to add it every time? We could even use page meta data to control what text appears for `more` in the future if different values are needed other than just `more`.

Reply:
There might be cards where the short description is the same as long description.
For that reason, I've added `...more` to distinguish between the the cards where they are different.